plugins/c8y_firmware_plugin/src/tests.rs

```diff
@@ -0,0 +1,775 @@
+use super::*;
+use crate::FirmwareManager;
+use assert_json_diff::assert_json_include;
+use c8y_api::http_proxy::MockC8YHttpProxy;
+use c8y_api::smartrest::error::SMCumulocityMapperError;
+use mockall::predicate;
+use mqtt_tests::with_timeout::WithTimeout;
+use mqtt_tests::StreamExt;
+use serde_json::json;
+use sha256::digest;
+use tedge_test_utils::fs::TempTedgeDir;
+
+const TEST_TIMEOUT_MS: Duration = Duration::from_millis(5000);
+const DEFAULT_REQUEST_TIMEOUT_SEC: Duration = Duration::from_secs(3600);
+
+const CHILD_DEVICE_ID: &str = "child-device";
+const FIRMWARE_NAME: &str = "fw-name";
+const FIRMWARE_VERSION: &str = "fw-version";
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+#[serial_test::serial]
+async fn handle_request_child_device() -> anyhow::Result<()> {
+    let mut tmp_dir = TempTedgeDir::new();
+    create_required_directories(&mut tmp_dir);
```

Comment:
~~Why not create these directories as part of the `start_firmware_manager` function itself?~~

Reply:
Got the answer to that from tests like `handle_request_dir_cache_not_found` where you want to pick and choose the directories that are created.

Reply:
Exactly!